docs/development/pigs/pig-016.rst

```diff
@@ -0,0 +1,215 @@
+in gammapy.image, and it's not clear what it's scope is.
+
+So we propose to clean this up, roughly like this:
+
+- asmooth.py and utils.py -> move to gammapy.cube
+- measure.py and profile.py -> replace with docs example or move to maps or cube
+- plotting.py -> replace with docs example or move to maps
+
+Clean up ``gammapy.utils``
+--------------------------
+
+Above we propose to move gammapy.utils.fitting, the biggest piece of
+gammapy.utils to gammapy.modeling. We still need and want to keep gammapy.utils,
+as a lowest-level layer in Gammapy where we write adapters and helpers for
+functionality e.g. from Numpy and Astropy. Mostly this is for usage within
+Gammapy, users very rarely need to import from gammapy.utils.
+
+Checking via ``ack gammapy.utils tutorials | egrep -v fitting`` we see that
+currently only energy_logspace, sqrt_space, sample_powerlaw and
+SphericalCircleSkyRegion are used from end-user documentation, i.e. very little.
+
+We propose to clean up gammapy.utils, e.g. gammapy/utils/coordinates.py still
+contains coordinate helper functions that were added before astropy.coordinates
+existed. We do not detail the cleanup here, this will be discussed on a
+case-by-case basis. After that we will review what is left and decide whether we
+keep or completely remove gammapy.utils from the Gammapy docs, i.e. if it's
+public API or not.
+
+Outlook
+=======
+
+During discussion of this PIG, several other ideas surfaced and were considered.
+We don't propose those changes here and for v1.0, in some cases because
+discussion didn't conclude on whether it's a good idea or not, and in some cases
+because introducing the change would require a detailed dedicated PIG with a
+feasability exploration as in `GH 2290`_ and then significant extra
+developments. We do expect that some of these changes will be made in the
+future (and some won't).
+
+Generalising maps to allow maps without a WCS and spatial axes, and / or using
+it everywhere in Gammapy (for IRFs, spectra, lightcurves) could be nice.
+Possibly ALMA does this, all their data is in 4D containers with lon, lat,
+wavelength, time, and e.g. a 1D spectrum would have only one bin in lon, lat,
+time.
+
+CTA IRFs and other CTA data level 3 classes might be maintained in ctapipe or
+somehow shared between Gammapy and ctapipe.
+
+gammapy.spectrum will likely become higher-level than gammapy.cube, i.e. 1D
+spectral analysis will be based on 3D maps analysis more.
+
+gammapy.detect will likely grow map analysis and modeling based algorithms, i.e.
+depend on gammapy.cube and the rest of the Gammapy package
+
+gammapy.astro could grow (merge in Naima) or be split out from Gammapy.
+
+gammapy.cube could be renamed to gammapy.map_analysis or gammapy.scripts to
+gammapy.analysis or similar changes depending on how their purpose and scope
+changes in the future datasets could be grouped in gammapy.datasets, if they are
```

Comment:
purpose and scope change. In the future, datasets could be grouped in

Reply:
fixed.